src/cryptography/hazmat/backends/openssl/encode_asn1.py

```diff
@@ -385,8 +391,12 @@ def _encode_subject_key_identifier(backend, ski):
 
 
 def _encode_general_name(backend, name):
+    gn = backend._lib.GENERAL_NAME_new()
+    return _encode_general_name_preallocated(backend, name, gn)
```

Comment:
I don't think it makes sense for `_encode_general_name_preallocated` to return anything

Reply:
Oops yeah it should just call this and then `return gn`